### ISSUE TITLE
Enum mismatch error not reported when more than one invalid type errors

### ISSUE THREAD

**Issue by gm1190** [zaggino/z-schema#230]
```
var ZSchema = require("z-schema");
var options = {noExtraKeywords: true};
var validator = new ZSchema(options);

var schema =
    {
        "title": "Policy structure schema",
        "description": "Defines the policy structure and properties description",
        "type": "object",
        "properties": {
            "version": {
                "title": "Version",
                "description": "Policy version",
                "minLength": 1,
                "maxLength": 256,
                "type": "string"
            },
            "name": {
                "type": "string",
                "title": "Name",
                "description": "Policy name",
                "minLength": 1,
                "maxLength": 256
            },
            "rules": {
                "title": "Rules",
                "description": "Policy rules",
                "type": "array",
                "minItems": 1,
                "maxItems": 256,
                "items": {
                    "$ref": "#/definitions/Rule"
                }
            }
        },
        "additionalProperties": true,
        "required": [
            "version",
            "rules"
        ],
        "definitions": {
            "ConditionItem": {
                "title": "Condition items",
                "description": "Defines the structure of conditions items",
                "type": "object",
                "minProperties": 1,
                "maxProperties": 256
            },
            "Condition": {
                "title": "Conditions",
                "minProperties": 1,
                "maxProperties": 256,
                "description": "Policy conditions",
                "type": "object",
                "anyOf": [
                    {
                        "required": [
                            "NumericEquals"
                        ]
                    },
                    {
                        "required": [
                            "NumericNotEquals"
                        ]
                    },
                    {
                        "required": [
                            "NumericLessThan"
                        ]
                    },
                    {
                        "required": [
                            "NumericGreaterThanEquals"
                        ]
                    },
                    {
                        "required": [
                            "NumericGreaterThan"
                        ]
                    },
                    {
                        "required": [
                            "NumericLessThanEquals"
                        ]
                    },
                    {
                        "required": [
                            "DateEquals"
                        ]
                    },
                    {
                        "required": [
                            "DateNotEquals"
                        ]
                    },
                    {
                        "required": [
                            "DateLessThan"
                        ]
                    },
                    {
                        "required": [
                            "DateGreaterThanEquals"
                        ]
                    },
                    {
                        "required": [
                            "DateGreaterThan"
                        ]
                    },
                    {
                        "required": [
                            "DateLessThanEquals"
                        ]
                    },
                    {
                        "required": [
                            "BinaryEquals"
                        ]
                    },
                    {
                        "required": [
                            "BinaryNotEquals"
                        ]
                    },
                    {
                        "required": [
                            "Null"
                        ]
                    },
                    {
                        "required": [
                            "StringEquals"
                        ]
                    },
                    {
                        "required": [
                            "StringNotEquals"
                        ]
                    },
                    {
                        "required": [
                            "StringEqualsIgnoreCase"
                        ]
                    },
                    {
                        "required": [
                            "StringNotEqualsIgnoreCase"
                        ]
                    },
                    {
                        "required": [
                            "StringLike"
                        ]
                    },
                    {
                        "required": [
                            "StringNotLike"
                        ]
                    },
                    {
                        "required": [
                            "Bool"
                        ]
                    },
                    {
                        "required": [
                            "NumericEqualsIfExists"
                        ]
                    },
                    {
                        "required": [
                            "ForAllValues:NumericEquals"
                        ]
                    },
                    {
                        "required": [
                            "ForAnyValue:NumericEquals"
                        ]
                    },
                    {
                        "required": [
                            "NumericNotEqualsIfExists"
                        ]
                    },
                    {
                        "required": [
                            "ForAllValues:NumericNotEquals"
                        ]
                    },
                    {
                        "required": [
                            "ForAnyValue:NumericNotEquals"
                        ]
                    },
                    {
                        "required": [
                            "NumericLessThanIfExists"
                        ]
                    },
                    {
                        "required": [
                            "ForAllValues:NumericLessThan"
                        ]
                    },
                    {
                        "required": [
                            "ForAnyValue:NumericLessThan"
                        ]
                    },
                    {
                        "required": [
                            "NumericGreaterThanEqualsIfExists"
                        ]
                    },
                    {
                        "required": [
                            "ForAllValues:NumericGreaterThanEquals"
                        ]
                    },
                    {
                        "required": [
                            "ForAnyValue:NumericGreaterThanEquals"
                        ]
                    },
                    {
                        "required": [
                            "NumericGreaterThanIfExists"
                        ]
                    },
                    {
                        "required": [
                            "ForAllValues:NumericGreaterThan"
                        ]
                    },
                    {
                        "required": [
                            "ForAnyValue:NumericGreaterThan"
                        ]
                    },
                    {
                        "required": [
                            "NumericLessThanEqualsIfExists"
                        ]
                    },
                    {
                        "required": [
                            "ForAllValues:NumericLessThanEquals"
                        ]
                    },
                    {
                        "required": [
                            "ForAnyValue:NumericLessThanEquals"
                        ]
                    },
                    {
                        "required": [
                            "DateEqualsIfExists"
                        ]
                    },
                    {
                        "required": [
                            "ForAllValues:DateEquals"
                        ]
                    },
                    {
                        "required": [
                            "ForAnyValue:DateEquals"
                        ]
                    },
                    {
                        "required": [
                            "DateNotEqualsIfExists"
                        ]
                    },
                    {
                        "required": [
                            "ForAllValues:DateNotEquals"
                        ]
                    },
                    {
                        "required": [
                            "ForAnyValue:DateNotEquals"
                        ]
                    },
                    {
                        "required": [
                            "DateLessThanIfExists"
                        ]
                    },
                    {
                        "required": [
                            "ForAllValues:DateLessThan"
                        ]
                    },
                    {
                        "required": [
                            "ForAnyValue:DateLessThan"
                        ]
                    },
                    {
                        "required": [
                            "DateGreaterThanEqualsIfExists"
                        ]
                    },
                    {
                        "required": [
                            "ForAllValues:DateGreaterThanEquals"
                        ]
                    },
                    {
                        "required": [
                            "ForAnyValue:DateGreaterThanEquals"
                        ]
                    },
                    {
                        "required": [
                            "DateGreaterThanIfExists"
                        ]
                    },
                    {
                        "required": [
                            "ForAllValues:DateGreaterThan"
                        ]
                    },
                    {
                        "required": [
                            "ForAnyValue:DateGreaterThan"
                        ]
                    },
                    {
                        "required": [
                            "DateLessThanEqualsIfExists"
                        ]
                    },
                    {
                        "required": [
                            "ForAllValues:DateLessThanEquals"
                        ]
                    },
                    {
                        "required": [
                            "ForAnyValue:DateLessThanEquals"
                        ]
                    },
                    {
                        "required": [
                            "BinaryEqualsIfExists"
                        ]
                    },
                    {
                        "required": [
                            "ForAllValues:BinaryEquals"
                        ]
                    },
                    {
                        "required": [
                            "ForAnyValue:BinaryEquals"
                        ]
                    },
                    {
                        "required": [
                            "BinaryNotEqualsIfExists"
                        ]
                    },
                    {
                        "required": [
                            "ForAllValues:BinaryNotEquals"
                        ]
                    },
                    {
                        "required": [
                            "ForAnyValue:BinaryNotEquals"
                        ]
                    },
                    {
                        "required": [
                            "NullIfExists"
                        ]
                    },
                    {
                        "required": [
                            "ForAllValues:Null"
                        ]
                    },
                    {
                        "required": [
                            "ForAnyValue:Null"
                        ]
                    },
                    {
                        "required": [
                            "StringEqualsIfExists"
                        ]
                    },
                    {
                        "required": [
                            "ForAllValues:StringEquals"
                        ]
                    },
                    {
                        "required": [
                            "ForAnyValue:StringEquals"
                        ]
                    },
                    {
                        "required": [
                            "StringNotEqualsIfExists"
                        ]
                    },
                    {
                        "required": [
                            "ForAllValues:StringNotEquals"
                        ]
                    },
                    {
                        "required": [
                            "ForAnyValue:StringNotEquals"
                        ]
                    },
                    {
                        "required": [
                            "StringEqualsIgnoreCaseIfExists"
                        ]
                    },
                    {
                        "required": [
                            "ForAllValues:StringEqualsIgnoreCase"
                        ]
                    },
                    {
                        "required": [
                            "ForAnyValue:StringEqualsIgnoreCase"
                        ]
                    },
                    {
                        "required": [
                            "StringNotEqualsIgnoreCaseIfExists"
                        ]
                    },
                    {
                        "required": [
                            "ForAllValues:StringNotEqualsIgnoreCase"
                        ]
                    },
                    {
                        "required": [
                            "ForAnyValue:StringNotEqualsIgnoreCase"
                        ]
                    },
                    {
                        "required": [
                            "StringLikeIfExists"
                        ]
                    },
                    {
                        "required": [
                            "ForAllValues:StringLike"
                        ]
                    },
                    {
                        "required": [
                            "ForAnyValue:StringLike"
                        ]
                    },
                    {
                        "required": [
                            "StringNotLikeIfExists"
                        ]
                    },
                    {
                        "required": [
                            "ForAllValues:StringNotLike"
                        ]
                    },
                    {
                        "required": [
                            "ForAnyValue:StringNotLike"
                        ]
                    },
                    {
                        "required": [
                            "BoolIfExists"
                        ]
                    },
                    {
                        "required": [
                            "ForAllValues:Bool"
                        ]
                    },
                    {
                        "required": [
                            "ForAnyValue:Bool"
                        ]
                    }
                ],
                "properties": {
                    "NumericEquals": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "NumericNotEquals": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "NumericLessThan": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "NumericGreaterThanEquals": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "NumericGreaterThan": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "NumericLessThanEquals": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "DateEquals": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "DateNotEquals": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "DateLessThan": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "DateGreaterThanEquals": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "DateGreaterThan": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "DateLessThanEquals": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "BinaryEquals": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "BinaryNotEquals": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "Null": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "StringEquals": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "StringNotEquals": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "StringEqualsIgnoreCase": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "StringNotEqualsIgnoreCase": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "StringLike": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "StringNotLike": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "Bool": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "NumericEqualsIfExists": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAllValues:NumericEquals": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAnyValue:NumericEquals": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "NumericNotEqualsIfExists": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAllValues:NumericNotEquals": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAnyValue:NumericNotEquals": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "NumericLessThanIfExists": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAllValues:NumericLessThan": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAnyValue:NumericLessThan": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "NumericGreaterThanEqualsIfExists": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAllValues:NumericGreaterThanEquals": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAnyValue:NumericGreaterThanEquals": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "NumericGreaterThanIfExists": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAllValues:NumericGreaterThan": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAnyValue:NumericGreaterThan": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "NumericLessThanEqualsIfExists": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAllValues:NumericLessThanEquals": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAnyValue:NumericLessThanEquals": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "DateEqualsIfExists": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAllValues:DateEquals": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAnyValue:DateEquals": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "DateNotEqualsIfExists": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAllValues:DateNotEquals": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAnyValue:DateNotEquals": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "DateLessThanIfExists": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAllValues:DateLessThan": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAnyValue:DateLessThan": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "DateGreaterThanEqualsIfExists": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAllValues:DateGreaterThanEquals": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAnyValue:DateGreaterThanEquals": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "DateGreaterThanIfExists": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAllValues:DateGreaterThan": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAnyValue:DateGreaterThan": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "DateLessThanEqualsIfExists": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAllValues:DateLessThanEquals": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAnyValue:DateLessThanEquals": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "BinaryEqualsIfExists": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAllValues:BinaryEquals": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAnyValue:BinaryEquals": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "BinaryNotEqualsIfExists": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAllValues:BinaryNotEquals": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAnyValue:BinaryNotEquals": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "NullIfExists": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAllValues:Null": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAnyValue:Null": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "StringEqualsIfExists": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAllValues:StringEquals": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAnyValue:StringEquals": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "StringNotEqualsIfExists": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAllValues:StringNotEquals": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAnyValue:StringNotEquals": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "StringEqualsIgnoreCaseIfExists": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAllValues:StringEqualsIgnoreCase": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAnyValue:StringEqualsIgnoreCase": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "StringNotEqualsIgnoreCaseIfExists": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAllValues:StringNotEqualsIgnoreCase": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAnyValue:StringNotEqualsIgnoreCase": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "StringLikeIfExists": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAllValues:StringLike": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAnyValue:StringLike": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "StringNotLikeIfExists": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAllValues:StringNotLike": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAnyValue:StringNotLike": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "BoolIfExists": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAllValues:Bool": {
                        "$ref": "#/definitions/ConditionItem"
                    },
                    "ForAnyValue:Bool": {
                        "$ref": "#/definitions/ConditionItem"
                    }
                },
                "additionalProperties": false
            },
            "Rule": {
                "type": "object",
                "additionalProperties": true,
                "title": "Rules item",
                "description": "Policy rules item",
                "required": [
                    "effect",
                    "principals",
                    "actions",
                    "resources"
                ],
                "properties": {
                    "effect": {
                        "title": "Effect",
                        "description": "Policy effect",
                        "type": "string",
                        "enum": [
                            "Allow",
                            "Deny"
                        ]
                    },
                    "actions": {
                        "title": "Actions",
                        "description": "Policy actions",
                        "type": "array",
                        "minItems": 1,
                        "maxItems": 256,
                        "items": {
                            "title": "Actions Items",
                            "description": "Policy actions item",
                            "minLength": 1,
                            "maxLength": 256,
                            "type": "string"
                        }
                    },
                    "resources": {
                        "title": "Resources",
                        "description": "Policy resources",
                        "type": "array",
                        "minItems": 1,
                        "maxItems": 256,
                        "items": {
                            "title": "Resources Items",
                            "description": "Policy resources item",
                            "minLength": 1,
                            "maxLength": 256,
                            "type": "string"
                        }
                    },
                    "principals": {
                        "title": "Principals",
                        "description": "Policy principals",
                        "type": "array",
                        "minItems": 1,
                        "maxItems": 256,
                        "items": {
                            "title": "Principals Items",
                            "description": "Policy principals item",
                            "minLength": 1,
                            "maxLength": 256,
                            "type": "string"
                        }
                    },
                    "condition": {
                        "$ref": "#/definitions/Condition"
                    }
                }
            }
        }
    }

;

var data =  {
    rules: [{
        'resources': 'BAD_VALUE',
        'effect': 'BAD_VALUE',
        'actions': 'BAD_VALUE',
        'principals': [ 'GOOD_VALUE' ]
    }],
    version: 'v1'
};

var valid = validator.validate(data, schema);

var error = validator.getLastError();

```
expecting 3 errors but getting 2 errors with above code
Expect output: code:INVALID_TYPE
                         code:INVALID_TYPE
                         code:ENUM_MISMATCH

Actual output :code:INVALID_TYPE
                         code:INVALID_TYPE

Can you please help me with this issue?

it is working fine with npm version 3.19.1

once i updated with 3.25.1(latest) version it is giving issue

**Comment by zaggino** [comment]
Closing as outdated and I believe fixed in current version.
